File: main/src/model/waterGoal.java
package model;

public class waterGoal extends Goal{
    private int currentAmount;
    private final int goalAmount;
    private final String goalType = "Litres";

    public waterGoal(int goalAmount) {
        super();
        this.goalAmount = goalAmount;
        currentAmount = 0;
    }

    public String getGoalType() {
        return goalType;
    }

    public int getCurrentAmount() {
        return currentAmount;
    }

    public int getGoalAmount() {
        return goalAmount;
    }

    public void setCurrentAmount(int currentAmount) {
        this.currentAmount = currentAmount;
    }



}

File: sustainagoal/src/model/waterGoal.ts
import Goal from "./Goal";

export default class waterGoal extends Goal {
    // private  currentAmount: number = 0;
    private readonly goalAmount: number = 0;
    private  goalType: string = "Liters";

    constructor(goalAmount: number) {
        super();
        this.goalAmount = goalAmount;
    }

    getGoalType(): string {
        return this.goalType;
    }

    // getCurrentAmount(): number {
    //     return this.currentAmount;
    // }

    getGoalAmount(): number {
        return this.goalAmount;
    }

    // setCurrentAmount(currentAmount: number) {
    //     this.currentAmount = currentAmount;
    // }

}